### DISCUSSION TITLE
No proxying after Windows restart

### DISCUSSION THREAD

**Discussion by tms320** [wiresock/proxifyre#64] [Q&A] [answered]
Hello!
I installed ProxiFyre as Windows 11 service. My config:
```
{
 "logLevel": "None",
 "proxies": [
         {
         "appNames": ["browser.exe"],
         "socks5ProxyEndpoint": "IP_ADDR:PORT",
         "username": "LOGIN",
         "password": "PASSWORD",
         "supportedProtocols": ["TCP", "UDP"]
         }
     ]
}
```
(I proxy Yandex.Browser)
Proxying works fine. Then I restart Windows. After logon I see that ProxiFyre service is running. Then I launch Yandex.Browser. But there is no proxying. I restart ProxiFyre service (sometimes several times) and it helps - proxying starts to work. Sometimes restarting ProxiFyre service doesn't help, but restarting Yandex.Browser helps.

**Comment by wiresock** [comment, marked as answer]
Hello!

Thanks for the detailed description.

The issue you're experiencing is likely due to the timing of service startup during Windows boot. When the ProxiFyre service starts automatically after a reboot, DNS resolution may not yet be fully available. As a result, the service fails to resolve the proxy server's hostname, and the proxy is not added correctly. This leads to the configuration not being fully applied, which explains why the proxying does not work until you restart the service.

**Recommended fix:**
Instead of using a hostname (e.g., `proxy.example.com`), please try using the **IP address** of the proxy server directly (e.g., `192.0.2.123:1080`). This way, the service does not depend on DNS at startup and can apply the configuration correctly right away.

Let us know if that helps!
